streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -272,11 +285,20 @@ private void commitOffsets() {
                 consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset));
                 stateMgr.putOffsetLimit(partition, offset);
             }
-            try {
-                consumer.commitSync(consumedOffsetsAndMetadata);
-            } catch (final CommitFailedException cfe) {
-                log.warn("{} Failed offset commits: {} ", logPrefix, consumedOffsetsAndMetadata);
-                throw cfe;
+
+            if (eosEnabled) {
+                producer.sendOffsetsToTransaction(consumedOffsetsAndMetadata, applicationId);
+                producer.commitTransaction();
+                if (startNewTransaction) {
+                    producer.beginTransaction();
+                }
+            } else {
+                try {
+                    consumer.commitSync(consumedOffsetsAndMetadata);
+                } catch (final CommitFailedException e) {
+                    log.warn("{} Failed offset commits {} due to {}", logPrefix, consumedOffsetsAndMetadata, e.getMessage());
```

Comment:
`CommitFailedException` can only be thrown if the consumer has fallen out of the group, we should handle it inside Streams than throw the exception all the way up.